### DISCUSSION TITLE
Coreference does work when the tokenisation isn't correct?

### DISCUSSION THREAD

**Discussion by KennethEnevoldsen** [explosion/spaCy#12532] [Help: Coding & Implementations] [answered]
So I am currently trying to train a Danish coreference model. It took some time to get the dataset in the correct format (I at least think that it is). Feel free to check it out [here](https://github.com/centre-for-humanities-computing/DaCy/tree/training-v0.2.0/training/v0.2.0).

When running the following training command:
```
spacy train configs/cluster.cfg --output training/cluster --paths.train corpus/cdt/train.spacy --paths.dev corpus/cdt/dev.spacy --nlp.lang=da
```

I get the following error:
```
IndexError: Misalignment in Coref. Head token has no match in training doc.
```

Running it in debug mode I find that the error is indeed correct:
In the text:

> Boksen er et resultat af et af rockhistoriens mest ambitiøse plade-projekter: Den rummer den komplette række af originale indspilninger fra Elvis'produktion i 50'erne

The tokens `Elvis'produktion` is annotated as two tokens, but is tokenized in the training as one token. Thus the head token has no match.

I can naturally go in and resolve these error in the dataset, but that seems problematic especially if you want to use annotations on noisy texts. Alternatively one could train assuming the "gold" tokens, but that will overestimate the performance of the model (and the model does not 'learn' to deal with the errors of the tokenizer).

I am unsure what the best solution is for this, I would love to hear a second opinion

**Comment by svlandeg** [comment, marked as answer]
I think this has been answered in https://github.com/explosion/spaCy/discussions/12533#discussioncomment-5634802, right?

**Reply by KennethEnevoldsen** [reply]
Indeed!
